Modules/TOF/src/CheckDRMDiagnostics.cxx

```diff
@@ -0,0 +1,92 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+///
+/// \file   CheckDRMDiagnostics.cxx
+/// \author Nicolo' Jacazio, Pranjal Sarma
+/// \brief  Checker dedicated to the study of low level raw data diagnostics words
+///
+
+// QC
+#include "TOF/CheckDRMDiagnostics.h"
+#include "QualityControl/MonitorObject.h"
+#include "QualityControl/Quality.h"
+#include "QualityControl/QcInfoLogger.h"
+
+// ROOT
+#include <TH1.h>
+#include <TH2.h>
+#include <TPaveText.h>
+#include <TList.h>
+
+using namespace std;
+
+namespace o2::quality_control_modules::tof
+{
+
+void CheckDRMDiagnostics::configure(std::string) {}
+
+Quality CheckDRMDiagnostics::check(std::map<std::string, std::shared_ptr<MonitorObject>>* moMap)
+{
+
+  Quality result = Quality::Null;
+  ILOG(Info, Support) << "Checking quality of diagnostic words";
+
+  for (auto& [moName, mo] : *moMap) {
+    (void)moName;
+    if (mo->getName() == "DRMCounter") {
+      auto* h = dynamic_cast<TH2F*>(mo->getObject());
+//      if (h->GetEntries() == 0) {
+//        result = Quality::Medium;
+
+	for (int i = 1; i < h->GetNbinsX(); i++) {
+	for (int j = 1; j < h->GetNbinsY(); j++) {
+          const float content = h->GetBinContent(i, j);
```

Comment:
@pranjalsarma perfect! Here we can check that the histogram does not have bins with errors inside. The meaning of each bin should be this one: https://github.com/AliceO2Group/AliceO2/blob/dev/DataFormats/Detectors/TOF/include/DataFormatsTOF/CompressedDataFormat.h#L118-L134 I think that we can skip the first bin (HAS_DATA) because this is fine.